tests/test_views.py

```diff
@@ -18,3 +22,76 @@ def setUp(self):
     def test_begin_view(self):
         response = self.client.get(reverse('social:begin', kwargs={'backend': 'facebook'}))
         self.assertEqual(response.status_code, 302)
+
+class TestGetSessionTimeout(TestCase):
+    """
+    Ensure that the branching logic of get_session_timeout behaves as expected.
+    """
+
+    def setUp(self):
+        self.social_user = mock.MagicMock()
+        self.social_user.expiration_datetime.return_value = None
+        super(TestGetSessionTimeout, self).setUp()
+
+    def set_user_expiration(self, seconds, days=0):
+        self.social_user.expiration_datetime.return_value = mock.MagicMock(
+            seconds=seconds,
+            days=days,
+        )
+
+    def test_no_max_expiration_disabled(self):
+        self.set_user_expiration(60)
+        expiration_length = get_session_timeout(
+            self.social_user,
+            enable_session_expiration=False
+        )
+        self.assertIs(expiration_length, None)
+
+    def test_max_present_expiration_disabled(self):
+        expiration_length = get_session_timeout(
+            self.social_user,
+            enable_session_expiration=False,
+            max_session_length=60
+        )
+        self.assertEqual(expiration_length, 60)
+
+    def test_user_has_session_length_no_max(self):
+        self.set_user_expiration(60)
+        expiration_length = get_session_timeout(
+            self.social_user,
+            enable_session_expiration=True
+        )
+        self.assertEqual(expiration_length, 60)
+
+    def test_user_has_session_length_larger_max(self):
+        self.set_user_expiration(60)
+        expiration_length = get_session_timeout(
+            self.social_user,
+            enable_session_expiration=True,
+            max_session_length=90
+        )
+        self.assertEqual(expiration_length, 60)
+
+    def test_user_has_session_length_smaller_max(self):
+        self.set_user_expiration(60)
+        expiration_length = get_session_timeout(
+            self.social_user,
+            enable_session_expiration=True,
+            max_session_length=30
+        )
+        self.assertEqual(expiration_length, 30)
+
+    def test_user_has_no_session_length_with_max(self):
+        expiration_length = get_session_timeout(
+            self.social_user,
+            enable_session_expiration=True,
+            max_session_length=60
+        )
+        self.assertEqual(expiration_length, 60)
+
+    def test_neither_explicit_nor_max(self):
```

Comment:
@haikuginger Suggestions for renaming these tests to make the naming scheme even more consistent:

* `test_no_max_expiration_disabled` ⟶ `test_expiration_disabled_no_max`
* `test_max_present_expiration_disabled` ⟶ `test_expiration_disabled_with_max`
* `test_neither_explicit_nor_max` ⟶ `test_user_has_no_session_length_no_max`